### PR TITLE
Feature/harmonize batch type signature

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
+ extra method on batch_constant that simplifies some common operation.

File: docs/source/api/batch_manip.rst [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 
-
-In the specific case when one needs to conditionnaly increment or decrement a
-batch based on a mask, :cpp:func:`incr_if` and
-:cpp:func:`decr_if` provide specialized version.
+                In the specific case when one needs to conditionnaly increment
+    or decrement a
+    batch based on a mask,
+    :
+cpp : func :`incr_if` and : cpp : func :`decr_if` provide specialized version.
```
Comment (JohanMabille) [inline docs/source/api/batch_manip.rst:27]: I'm afraid all these formatting changes break the rendering of rst.

**Review by JohanMabille — COMMENTED** [review]
Except for the rst changes (which I'm not sure), LGTM.

File: docs/source/api/batch_manip.rst (hunk 1 above)
Comment (serge-sans-paille) [reply docs/source/api/batch_manip.rst]: Yeah, that was an erroneous clang-format call, thanks for spotting this

**Comment by serge-sans-paille** [comment]
Thanks for the reviews o/

**Pull request merged**